### PR TITLE
Avoid error on order create when listPrice rule is empty

### PR DESCRIPTION
**Pull request by Isengo1989**
### 1. Why is this change necessary?
**INFO**: This is a pickup from https://github.com/shopware/platform/pull/2654 - since it was closed I reopend it. Hope this is okay this way

to prevent a WriteException when persisting an order with a promotion using the LineItemListPriceRule with the empty-operator.

### 2. What does this change do, exactly?
I added a missing Constraint for the "amount" field

### 3. Describe each step to reproduce the issue or behaviour.

create a new promotion (no promotion code to apply automatically)
add a new discount, only applied to certain products => rule "Item with liste price => at least one => is empty"
apply to all items
percentage => 10
=> the goal is to create a promotion that automatically applies to all non-discounted products
go to checkout and order a non discounted product (make sure the promotion applies)
the order cannot be saved => an error is thrown (WriteException)

4. Please link to the relevant issues (if any).

### 5. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [x] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/workflow/2020-08-03-implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

<a href="https://gitpod.io/#https://github.com/shopware/platform/pull/2758"></a>

### CODE REVIEW

**Comment by Isengo1989** [comment]
Not sure why I get a platform error

https://github.com/shopware/platform/actions/runs/3226419147/jobs/5279911272#step:5:108

File: src/Core/Checkout/Cart/Rule/LineItemListPriceRule.php [hunk 1]
```diff
@@ -61,6 +62,10 @@ public function getConstraints(): array
         ];
 
         if ($this->operator === self::OPERATOR_EMPTY) {
+            $constraints['amount'] = new EqualTo([
+                'value' => 0,
+            ]);
```
Comment (dneustadt) [inline src/Core/Checkout/Cart/Rule/LineItemListPriceRule.php:67]: Generally speaking it is correct that the constraints should not include `amount` if the `empty` operator is used. If the rule is persisted via the administration `amount` should be unset if the operator is switched to `empty`. The real question is why the rule is persisted when persisting an order and why an amount is assigned when it previously wasn't. I feel like if that is the case, that's something that should be fixed at that point instead.

**Comment by dneustadt** [comment]
Hey, first of all, thanks for taking the time and effort to help solve this issue. This turned out to be a general issue with conditions allowing the `empty` operator. We'll fix the more deeply rooted cause of this issue. Unfortunately that means we're going to have to close your MR. Still your contribution is very much appreciated, as it brought attention to this issue and helped identifying the cause.

**Comment by Isengo1989** [comment]
@dneustadt  thanks for the update. No worries, I was just adding a test regarding that ticket anyways -> https://github.com/shopware/platform/pull/2654

Could you maybe link the issue so we can keep track on it ?

@supus  fyi

**Comment by dneustadt** [comment]
Sure. https://issues.shopware.com/issues/NEXT-23652

**Comment by supus** [comment]
@Isengo1989 sorry for the late reply. thank you so much for adding a test and tagging me 🙏

**Pull request closed without merging**